src/smc-webapp/smc-react.coffee

```diff
@@ -587,11 +588,35 @@ switch MODE
                 console.log r.displayName, "took", t1 - t0, "ms of time"
             return r
     when 'verbose'
+        renderings   = []
+        time_window  = 10 * 1000 # we take the last ~10 seconds into account
         rclass = (x) ->
             x._render = x.render
             x.render = () ->
-                console.log x.displayName
-                return @_render()
+                # measure time
+                t0 = performance.now()
```

Comment:
This is much, much more complicated than any of the other profiling code I added.   Can you put it in another file?  Also, make sure it doesn't just reproduce https://reactjs.org/docs/perf.html ?